library/c/sandesh.h

```diff
@@ -25,6 +25,52 @@ extern "C" {
 #include <Wsk.h>
 #include <Ws2ipdef.h>
 
+static void *
+s_malloc(unsigned int size)
+{
+    void *mem = ExAllocatePoolWithTag(NonPagedPool, size+1, 'RVCO');
```

Comment:
`size+1` will not allocate enough space for header and requested buffer size. `size` will be 4 bytes, so you need at least 4 bytes for header.